### ISSUE TITLE
Auto Terms not searching Titles

### ISSUE THREAD

**Issue by reekes** [TaxoPress/TaxoPress#1927]
## Expected Behavior

Creating a term with synonyms will include searching titles

## Current Behavior

I have the term (tag) "step by step" defined.

I have posts with "Step-By-Step" in their titles.

I cannot add "step-by-step" as a synonym. I'm assuming TaxoPress is going to ignore hyphans. Is that true? 

Yet TaxoPress doesn't tag the article with Step-By-Step in the title.

I also tried changing the term to be "step-by-step" and it still didn't tag the post with that term in the title.

## Possible Solution

none

## Steps to Reproduce (for bugs)

see above

## Context

I need to add tags to posts when my term appear in the title or content.

## Your Environment

* Version used: everything is up to date, TaxoPress is 3.11.0

**Comment by stevejburge** [comment]
Screenshot thanks to @reekes 

![PastedGraphic-1](https://github.com/TaxoPress/TaxoPress/assets/3868028/f50930d0-6383-4f5a-9d42-9ae663442dc5)

**Comment by stevejburge** [comment]
@reekes No problems with hyphens in my testing

@ojopaul Can we be more specific with the message? Could we specify which term is the problem? That might help narrow down the issue here.

**Comment by reekes** [comment]
Hmm, seems the failure case have this in the title: "Step-By-Step:  "

I think the colon at the end it causing the problem. Is that causing Auto Term to reject the match?
Is there a way to make matching ignore such punctuation?

**Comment by reekes** [comment]
nope, I found a post with the title "Step-by-Step Guide to Identify Inactive Users..."
and TaxoPress didn't match it with the term "Step-by-Step" (the exact name of the term is in the title)

**Comment by stevejburge** [comment]
@reekes We'll need more information on this one too, please. It appears to work:

![Screenshot 2023-10-31 at 8 52 33 AM](https://github.com/TaxoPress/TaxoPress/assets/3868028/0f6c4211-d27b-4b4c-bef8-4355a5d5531e)
![Screenshot 2023-10-31 at 8 53 57 AM](https://github.com/TaxoPress/TaxoPress/assets/3868028/ddf35dc7-5929-4fa8-afbf-a71ac285ea6a)

**Comment by reekes** [comment]
Well, here's what that looks like for me. I have a term "step by step" and here I'm showing a CPT with that in the title. We don't see the term being suggested.

![image](https://github.com/TaxoPress/TaxoPress/assets/18753948/fcb826df-02a9-42ae-9cb9-f342fc32b948)

**Comment by stevejburge** [comment]
@reekes Is the term "step by step" or "Step-by-Step?

**Comment by reekes** [comment]
I've tried both terms, and tried both options in the title. It never matches. I'm left to think it's about _not finding a term in the title_.
